builtin/credential/aws/path_login.go

```diff
@@ -800,12 +800,14 @@ func (b *backend) pathLoginUpdateEc2(ctx context.Context, req *logical.Request,
 		resp.Auth.Metadata["nonce"] = clientNonce
 	}
 
-	if roleEntry.MaxTTL > time.Duration(0) {
-		// Cap TTL to shortestMaxTTL
-		if resp.Auth.TTL > shortestMaxTTL {
-			resp.AddWarning(fmt.Sprintf("Effective TTL of '%s' exceeded the effective max_ttl of '%s'; TTL value is capped accordingly", (resp.Auth.TTL / time.Second), (shortestMaxTTL / time.Second)))
-			resp.Auth.TTL = shortestMaxTTL
-		}
+	// In this case no role value was set so pull in what will be assigned by
+	// Core for comparison
+	if resp.Auth.TTL == 0 {
+		resp.Auth.TTL = b.System().DefaultLeaseTTL()
+	}
+	if resp.Auth.TTL > shortestMaxTTL {
+		resp.Auth.TTL = shortestMaxTTL
+		resp.AddWarning(fmt.Sprintf("Effective TTL of '%s' exceeded the effective max_ttl of '%s'; TTL value is capped accordingly", (resp.Auth.TTL / time.Second), (shortestMaxTTL / time.Second)))
```

Comment:
Division by `time.Second` should be removed here to fix the print statement. (`(1 * time.Second) / time.Second` evaluates to 1 nanosecond) Same on line 1325.

Reply:
We display ttl/max_ttl in seconds to the user...so I think this behavior is actually what we want right?

Reply:
When we divide the TTL by `time.Second`, the print statement looks like this:

```
  * Effective TTL of '2.7648ms' exceeded the effective max_ttl of '600ns'; TTL
  value is capped accordingly
```

With the fix:

```
  * Effective TTL of '768h0m0s' exceeded the effective max_ttl of '10m0s'; TTL
  value is capped accordingly
```

Reply:
Ah, ok. That's because we need to print it as a digit, not a string.